File: DSA/Assignment_3/spiral.c
#include <stdio.h>
int main()
{
    int m,n;
    printf("Enter row and column size:");
    scanf("%d %d",&m,&n);
    int arr[m][n];
    
    int row_start=0,row_end=m-1;
    int col_start=0,col_end=n-1;
    int num=0;
    while(row_start<=row_end && col_start<=col_end)
    {
        for(int i=col_start;i<col_end;i++)
        arr[row_start][i]=++num;
        row_start++;
        for(int i=row_start;i<row_end;i++)
        arr[i][col_start]=++num;
        col_start++;
        if(row_start<=row_end){}
        
    }
    
    return 0;
}

File: DSA/Assignment_3/spiral.py
m=int(input("Enter row size:"))
n=int(input("Enter column size:"))
arr=[]
arr=[[0 for _ in range(n)] for _ in range(m)]

row_start, row_end=0, m
col_start, col_end=0, n
num=0

while(row_start<=row_end and col_start<=col_end):
    for i in range(col_start,col_end):
        num+=1
        arr[row_start][i]=num
    row_start+=1

    for i in range(row_start,row_end):
        num+=1
        arr[i][col_end-1]=num
    col_end-=1

    for i in range(col_end-1,col_start-1,-1):
        num+=1
        arr[row_end-1][i]=num
    row_end-=1

    for i in range(row_end-1,row_start-1,-1):
        num+=1
        arr[i][col_start]=num
    col_start+=1

print("Spiral Matrix:")
for i in arr:
    print(i)